From aeb92b0d7ea5373f033be06cdd23be389b94ecb6 Mon Sep 17 00:00:00 2001
From: "Liu,Dalei" <daleix.liu@intel.com>
Date: Mon, 11 Apr 2016 11:53:18 +0800
Subject: [PATCH 4/9] ov7251: add 640x480 100fps setting

Add sensor ov7251 640x480 100fps setting

Change-Id: I357658a6a41a4a68a24e895a82241680bcd9d569
Original-Tracked_On: #1504217386
Signed-off-by: Liu,Dalei <daleix.liu@intel.com>
Reviewed-on: https://icggerrit.ir.intel.com/113018
Reviewed-by: HaiqiangX Yao <haiqiangx.yao@intel.com>
Approver: Jian Xu Zheng <jian.xu.zheng@intel.com>
Validation-Android: Wei Tao <wei.tao@intel.com>
Build: Dessert M BTNS (SC)
Build: Dessert M BXT RVP (SC)
Build: Dessert M CHT FFD (SC)
Integrator: WeiX Meng <weix.meng@intel.com>
Reviewed-by: WeiX Meng <weix.meng@intel.com>
Tracked-On: https://jira01.devtools.intel.com/browse/OAM-28570
Signed-off-by: Tao Wei <wei.tao@intel.com>
Reviewed-on: https://android.intel.com:443/494418
---
 .../camera/drivers/media/i2c/ov7251.h              | 174 +++++++++++++++++++--
 1 file changed, 159 insertions(+), 15 deletions(-)

diff --git a/drivers/external_drivers/camera/drivers/media/i2c/ov7251.h b/drivers/external_drivers/camera/drivers/media/i2c/ov7251.h
index 4a52af2..6fa25ea 100644
--- a/drivers/external_drivers/camera/drivers/media/i2c/ov7251.h
+++ b/drivers/external_drivers/camera/drivers/media/i2c/ov7251.h
@@ -428,6 +428,150 @@ static const struct ov7251_reg ov7251_640_480_30fps[] = {
 	{OV7251_TOK_TERM, 0, 0},
 };
 
+static const struct ov7251_reg ov7251_640_480_100fps[] = {
+	{OV7251_8BIT, 0x100, 0x0},
+	{OV7251_8BIT, 0x3005, 0x8},
+	{OV7251_8BIT, 0x3012, 0xC0},
+	{OV7251_8BIT, 0x3013, 0xD2},
+	{OV7251_8BIT, 0x3014, 0x04},
+	{OV7251_8BIT, 0x3016, 0x10},
+	{OV7251_8BIT, 0x3017, 0x0},
+	{OV7251_8BIT, 0x3018, 0x0},
+	{OV7251_8BIT, 0x301A, 0x0},
+	{OV7251_8BIT, 0x301B, 0x0},
+	{OV7251_8BIT, 0x301C, 0x20},
+	{OV7251_8BIT, 0x3023, 0x5},
+	{OV7251_8BIT, 0x3037, 0xF0},
+	{OV7251_8BIT, 0x3098, 0x4},
+	{OV7251_8BIT, 0x3099, 0x32},
+	{OV7251_8BIT, 0x309A, 0x5},
+	{OV7251_8BIT, 0x309B, 0x4},
+	{OV7251_8BIT, 0x309D, 0x0},
+	{OV7251_8BIT, 0x30B0, 0xA},
+	{OV7251_8BIT, 0x30B1, 0x1},
+	{OV7251_8BIT, 0x30B3, 0x64},
+	{OV7251_8BIT, 0x30B4, 0x3},
+	{OV7251_8BIT, 0x30B5, 0x5},
+	{OV7251_8BIT, 0x3106, 0xDA},
+	{OV7251_8BIT, 0x3500, 0x00},
+	{OV7251_8BIT, 0x3501, 0x35},
+	{OV7251_8BIT, 0x3502, 0x20},
+	{OV7251_8BIT, 0x3503, 0x07},
+	{OV7251_8BIT, 0x3509, 0x10},
+	{OV7251_8BIT, 0x350A, 0x0},
+	{OV7251_8BIT, 0x350B, 0x0},
+	{OV7251_8BIT, 0x3600, 0x1C},
+	{OV7251_8BIT, 0x3602, 0x62},
+	{OV7251_8BIT, 0x3620, 0xB7},
+	{OV7251_8BIT, 0x3622, 0x4},
+	{OV7251_8BIT, 0x3626, 0x21},
+	{OV7251_8BIT, 0x3627, 0x30},
+	{OV7251_8BIT, 0x3630, 0x44},
+	{OV7251_8BIT, 0x3631, 0x35},
+	{OV7251_8BIT, 0x3634, 0x60},
+	{OV7251_8BIT, 0x3636, 0x0},
+	{OV7251_8BIT, 0x3662, 0x1},
+	{OV7251_8BIT, 0x3663, 0x70},
+	{OV7251_8BIT, 0x3664, 0xF0},
+	{OV7251_8BIT, 0x3666, 0xA},
+	{OV7251_8BIT, 0x3669, 0x1A},
+	{OV7251_8BIT, 0x366A, 0x0},
+	{OV7251_8BIT, 0x366B, 0x50},
+	{OV7251_8BIT, 0x3673, 0x1},
+	{OV7251_8BIT, 0x3674, 0xFF},
+	{OV7251_8BIT, 0x3675, 0x3},
+	{OV7251_8BIT, 0x3705, 0xC1},
+	{OV7251_8BIT, 0x3709, 0x40},
+	{OV7251_8BIT, 0x373C, 0x8},
+	{OV7251_8BIT, 0x3742, 0x0},
+	{OV7251_8BIT, 0x3757, 0xB3},
+	{OV7251_8BIT, 0x3788, 0x0},
+	{OV7251_8BIT, 0x37A8, 0x01},
+	{OV7251_8BIT, 0x37A9, 0xC0},
+	{OV7251_8BIT, 0x3800, 0x00},
+	{OV7251_8BIT, 0x3801, 0x4}, /* H CROP START: */
+	{OV7251_8BIT, 0x3802, 0x00},
+	{OV7251_8BIT, 0x3803, 0x4}, /* V CROP START:  */
+	{OV7251_8BIT, 0x3804, 0x02},
+	{OV7251_8BIT, 0x3805, 0x8B}, /* H CROP END: */
+	{OV7251_8BIT, 0x3806, 0x01},
+	{OV7251_8BIT, 0x3807, 0xEB}, /* V CROP END:  */
+	{OV7251_8BIT, 0x3808, 0x02},
+	{OV7251_8BIT, 0x3809, 0x80}, /* H OUTPUT SIZE: */
+	{OV7251_8BIT, 0x380A, 0x01},
+	{OV7251_8BIT, 0x380B, 0xE0}, /* V OUTPUT SIZE: */
+	{OV7251_8BIT, 0x380C, 0x03},
+	{OV7251_8BIT, 0x380D, 0xA0}, /* H TOTAL SIZE: */
+	{OV7251_8BIT, 0x380E, 0x02},
+	{OV7251_8BIT, 0x380F, 0x0E}, /* V TOTAL SIZE: */
+	{OV7251_8BIT, 0x3810, 0x00},
+	{OV7251_8BIT, 0x3811, 0x04}, /* H WINDOW OFFSET: 5 */
+	{OV7251_8BIT, 0x3812, 0x00},
+	{OV7251_8BIT, 0x3813, 0x01}, /* V WINDOW OFFSET:  */
+	{OV7251_8BIT, 0x3814, 0x11},
+	{OV7251_8BIT, 0x3815, 0x11},
+	{OV7251_8BIT, 0x3820, 0x40},
+	{OV7251_8BIT, 0x3821, 0x00},
+	{OV7251_8BIT, 0x382F, 0x0E},
+	{OV7251_8BIT, 0x3832, 0x00},
+	{OV7251_8BIT, 0x3833, 0x05},
+	{OV7251_8BIT, 0x3834, 0x00},
+	{OV7251_8BIT, 0x3835, 0x0C},
+	{OV7251_8BIT, 0x3837, 0x00},
+	{OV7251_8BIT, 0x3B80, 0x00},
+	{OV7251_8BIT, 0x3B81, 0xFF}, /* STROBE FRAME PATTERN */
+	{OV7251_8BIT, 0x3B82, 0x10},
+	{OV7251_8BIT, 0x3B83, 0x00},
+	{OV7251_8BIT, 0x3B84, 0x08},
+	{OV7251_8BIT, 0x3B85, 0x00},
+	{OV7251_8BIT, 0x3B86, 0x01},
+	{OV7251_8BIT, 0x3B87, 0x00},
+	{OV7251_8BIT, 0x3B88, 0x00},
+	{OV7251_8BIT, 0x3B89, 0x00},
+	{OV7251_8BIT, 0x3B8A, 0x00},
+	{OV7251_8BIT, 0x3B8B, 0x05},
+	{OV7251_8BIT, 0x3B8C, 0x00},
+	{OV7251_8BIT, 0x3B8D, 0x00},
+	{OV7251_8BIT, 0x3B8E, 0x00},
+	{OV7251_8BIT, 0x3B8F, 0x1A},
+	{OV7251_8BIT, 0x3B94, 0x05},
+	{OV7251_8BIT, 0x3B95, 0xF2},
+	{OV7251_8BIT, 0x3B96, 0x40},
+	{OV7251_8BIT, 0x3C00, 0x89},
+	{OV7251_8BIT, 0x3C01, 0x63},
+	{OV7251_8BIT, 0x3C02, 0x01},
+	{OV7251_8BIT, 0x3C03, 0x00},
+	{OV7251_8BIT, 0x3C04, 0x00},
+	{OV7251_8BIT, 0x3C05, 0x03},
+	{OV7251_8BIT, 0x3C06, 0x00},
+	{OV7251_8BIT, 0x3C07, 0x06},
+	{OV7251_8BIT, 0x3C0C, 0x01},
+	{OV7251_8BIT, 0x3C0D, 0xD0},
+	{OV7251_8BIT, 0x3C0E, 0x02},
+	{OV7251_8BIT, 0x3C0F, 0x0A},
+	{OV7251_8BIT, 0x4001, 0x42},
+	{OV7251_8BIT, 0x4004, 0x04}, /* NUMBER OF BLAND LINE USED */
+	{OV7251_8BIT, 0x4005, 0x00},
+	{OV7251_8BIT, 0x404E, 0x01},
+	{OV7251_8BIT, 0x4300, 0xFF},
+	{OV7251_8BIT, 0x4301, 0x00},
+	{OV7251_8BIT, 0x4501, 0x48},
+	{OV7251_8BIT, 0x4600, 0x00},
+	{OV7251_8BIT, 0x4601, 0x4E},
+	{OV7251_8BIT, 0x4801, 0x0F},
+	{OV7251_8BIT, 0x4806, 0x0F},
+	{OV7251_8BIT, 0x4819, 0xAA},
+	{OV7251_8BIT, 0x4823, 0x3E},
+	{OV7251_8BIT, 0x4837, 0x1F}, /* PCLK PERIOD */
+	{OV7251_8BIT, 0x4A0D, 0x00},
+	{OV7251_8BIT, 0x4A47, 0x7F},
+	{OV7251_8BIT, 0x4A49, 0xF0},
+	{OV7251_8BIT, 0x4A4B, 0x30},
+	{OV7251_8BIT, 0x5000, 0x85}, /* ISP CRTL00 */
+	{OV7251_8BIT, 0x5001, 0x80}, /* ISP CRTL01 */
+
+	{OV7251_TOK_TERM, 0, 0},
+};
 static const struct ov7251_reg ov7251_640_480_60fps[] = {
 	{OV7251_8BIT, 0x100, 0x0},
 	{OV7251_8BIT, 0x3005, 0x8},
@@ -864,24 +1008,24 @@ static const struct ov7251_reg ov7251_656_496_60fps[] = {
 };
 struct ov7251_resolution ov7251_res_preview[] = {
 	{
-		.desc = "ov7251_640_480_60fps",
+		.desc = "ov7251_640_480_100fps",
 		.width = 640,
 		.height = 480,
 		.pix_clk_freq = 73,
 		.used = 0,
 		.pixels_per_line = 928,
-		.lines_per_frame = 862,
+		.lines_per_frame = 522,
 		.bin_factor_x = 1,
 		.bin_factor_y = 1,
 		.bin_mode = 0,
 		.skip_frames = 3,
-		.regs = ov7251_640_480_60fps,
+		.regs = ov7251_640_480_100fps,
 		.mipi_freq = 422400,
 		.fps_option = {
 			{
-				.fps = 60,
+				.fps = 100,
 				.pixels_per_line = 928,
-				.lines_per_frame = 862,
+				.lines_per_frame = 522,
 				.regs = ov7251_640_480_60fps,
 			},
 			{
@@ -919,24 +1063,24 @@ struct ov7251_resolution ov7251_res_preview[] = {
 
 struct ov7251_resolution ov7251_res_still[] = {
 	{
-		.desc = "ov7251_640_480_60fps",
+		.desc = "ov7251_640_480_100fps",
 		.width = 640,
 		.height = 480,
 		.pix_clk_freq = 73,
 		.used = 0,
 		.pixels_per_line = 928,
-		.lines_per_frame = 862,
+		.lines_per_frame = 522,
 		.bin_factor_x = 1,
 		.bin_factor_y = 1,
 		.bin_mode = 0,
 		.skip_frames = 3,
-		.regs = ov7251_640_480_60fps,
+		.regs = ov7251_640_480_100fps,
 		.mipi_freq = 422400,
 		.fps_option = {
 			{
-				.fps = 60,
+				.fps = 100,
 				.pixels_per_line = 928,
-				.lines_per_frame = 862,
+				.lines_per_frame = 522,
 				.regs = ov7251_640_480_60fps,
 			},
 			{
@@ -973,24 +1117,24 @@ struct ov7251_resolution ov7251_res_still[] = {
 
 struct ov7251_resolution ov7251_res_video[] = {
 	{
-		.desc = "ov7251_640_480_60fps",
+		.desc = "ov7251_640_480_100fps",
 		.width = 640,
 		.height = 480,
 		.pix_clk_freq = 73,
 		.used = 0,
 		.pixels_per_line = 928,
-		.lines_per_frame = 862,
+		.lines_per_frame = 522,
 		.bin_factor_x = 1,
 		.bin_factor_y = 1,
 		.bin_mode = 0,
 		.skip_frames = 3,
-		.regs = ov7251_640_480_60fps,
+		.regs = ov7251_640_480_100fps,
 		.mipi_freq = 422400,
 		.fps_option = {
 			{
-				.fps = 60,
+				.fps = 100,
 				.pixels_per_line = 928,
-				.lines_per_frame = 862,
+				.lines_per_frame = 522,
 				.regs = ov7251_640_480_60fps,
 			},
 			{
-- 
1.9.1

